src/m365/purview/commands/auditlog/auditlog-list.ts

```diff
@@ -0,0 +1,244 @@
+import Auth from '../../../../Auth';
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import request, { CliRequestOptions } from '../../../../request';
+import { accessToken } from '../../../../utils/accessToken';
+import { validation } from '../../../../utils/validation';
+import O365MgmtCommand from '../../../base/O365MgmtCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  contentType: string;
+  startTime?: string;
+  endTime?: string;
+}
+
+class PurviewAuditLogListCommand extends O365MgmtCommand {
+  private readonly contentTypeOptions = ['AzureActiveDirectory', 'Exchange', 'SharePoint', 'General', 'DLP'];
+
+  public get name(): string {
+    return commands.AUDITLOG_LIST;
+  }
+
+  public get description(): string {
+    return 'List audit logs within your tenant';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['CreationTime', 'UserId', 'Operation', 'ObjectId'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        startTime: typeof args.options.startTime !== 'undefined',
+        endTime: typeof args.options.endTime !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '--contentType <contentType>',
+        autocomplete: this.contentTypeOptions
+      },
+      {
+        option: '--startTime [startTime]'
+      },
+      {
+        option: '--endTime [endTime]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (this.contentTypeOptions.indexOf(args.options.contentType) === -1) {
+          return `'${args.options.contentType}' is not a valid contentType value. Allowed values: ${this.contentTypeOptions}.`;
+        }
```

Comment:
I wouldn't restrict this option since we do not know which content types are going to be provisioned in the future. 

If we keep it, we have to update the command specs every time

Reply:
Hi Patrick

I don't know if these content type values will be changing anytime soon. Just checked the docs repo and these content types are there since the start of the repo 5 years ago.
Also we are mapping these user friendly content type names to their internal value (same file line 125):

```ts
const contentTypeValue = args.options.contentType === 'DLP' ? 'DLP.All' : 'Audit.' + args.options.contentType;
```

So in the unlikely event of new content type values, we have to update this piece of code again.

If you opt to let the user pass the internal content type value, I don't think we can expect our users to know the exact internal content type names. Since this requires you to read the office365 management API docs and makes it much less user friendly.

Reply:
Fair enough. 

Reasoning behind. I usually do not like to have commands limiting me if there is a chance that the service values could change. 

I leave it like this for now